pkg/backend/display/tree.go

```diff
@@ -193,11 +207,16 @@ func (r *treeRenderer) frame(locked, done bool) {
 	}
 	r.dirty = false
 
+	now := time.Now()
+	since := now.Sub(r.lastFrame)
+	r.lastFrame = now
```

Comment:
I think it might be simpler to track a deadline than a timeout on the status message.

In showStatusMessage, instead of recording statusMessageDuration, we would record statusMessageDeadline of when the message should stop showing

```go
func (r *treeRenderer) showStatusMessage(msg string, duration time.Duration) {
  // ...
  r.statusMessageDeadline = time.Now().Add(duration)
}

Then, if a status message was printed below around the "enforce timeout" bit:

```go
if statusMessage != "" {
  if time.Until(r.statusMessageDeadline) <= 0 {
    r.statusMessage, r.statusMessageDeadline = "", time.Time{}
  }
}
```

This will be a lot cheaper than tracking the time since each frame render (which, albeit, isn't that expensive) because it will hit time.Now and friends only if there was a status message to print in the first place—which is going to be rare.

Reply:
Simplicity I'm sympathetic to, but I don't find the cost argument persuasive--this won't be where we're spending our cycles. I like the suggestion to use a deadline, though--I'll make that change.

Reply:
I agree that the cost was negligible here, so yeah, that's more of a nice to have than the simpler state management which was the primary goal.